.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,58 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Publish Docker image
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    if: ${{ startsWith(github.repository, 'apache/') }}
+    runs-on: ubuntu-22.04
+    strategy:
+       matrix:
+         include:
+           - suffix-name: ''
+             opts: '--flink-provided --spark-provided --hive-provided'
+           - suffix-name: '-spark'
+             opts: '--flink-provided --hive-provided'
+           - suffix-name: '-flink'
+             opts: '--spark-provided --hive-provided'
+           - suffix-name: '-all'
+             opts: ''
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Login to Docker Hub
```

Comment:
Set up docker buildx before login as well, as multi-platform image building is one of the important foreseeable features.
```
      -
        name: Set up QEMU
        uses: docker/setup-qemu-action@v2
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2
```

Reply:
I think we can do that in followup once we decide to release multi-platform image. WDYT ?

Reply:
OK. These actions are now used in kyuubi workflows as well, just to mention it.
And btw, where to tell the arch of image from the dockfile or comments we are releasing?

Reply:
> where to tell the arch of image from the dockfile or comments we are releasing?

@bowenliang123  It's an option in dockerhub if we build some, e.g.

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/12025282/215972737-0800b15c-f3de-452f-ba1a-5e5c92161119.png">